main.tf

```diff
@@ -1,13 +1,37 @@
-#
-# Developer tips:
-#   - Below code should be replaced with the code for the root level module
-#
+# Using data block, retreiving the details of the subnets.
+data "ibm_is_subnet" "selected" {
+  count      = length(var.subnet_ids)
+  identifier = var.subnet_ids[count.index]
+}
+
+locals {
+  # Generating a list of bare metal server configurations
+  baremetal_server_list = flatten([
+    for idx in range(var.server_count) : {
+      prefix    = var.server_count == 1 ? var.prefix : "${var.prefix}-${idx}"
```

Comment:
I think that this line is what caused my destruction test to fail, because you drop the index off when it goes to size 1, which means when I scaled down my deployment it thought it had to replace my server that was left.